lib/view_component/instrumentation.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ViewComponent # :nodoc:
+  module Instrumentation
+    def self.included(mod)
+      mod.prepend(self)
+    end
+
+    def render_in(view_context, &block)
+      ActiveSupport::Notifications.instrument("!render.view_component", name: self.class.name, identifier: self.class.identifier) do
```

Comment:
The convention seems to be that the leading `!` means "*that the event is private of the framework and should not be used by applications and libraries*".

see https://twitter.com/rafaelfranca/status/1289978377550442496

I'm not sure it's the case here?

Reply:
Oh interesting, I had a different interpretation of the `!` prefix. I had made the suggestion to prefix it with a `!` for similar reasons that [this comment outlines](https://github.com/rails/rails/blob/91f0db1353e1b24e2d26ebbbadce510735f66848/actionview/lib/action_view/template.rb#L148-L150).

```
    # This method is instrumented as "!render_template.action_view". Notice that
    # we use a bang in this instrumentation because you don't want to
    # consume this in production. This is only slow if it's being listened to.
```

Instrumentation can become surprisingly slow when there are listeners, so I thought it made sense to try and discourage it where possible.